File: src/c/_0169_Majority_Element.c
int majorityElement(int *nums, int numsSize)
{
    int ans = nums[0], count = 1;

    for (int i = 1; i < numsSize; i++)
    {
        if (count == 0)
            ans = nums[i];

        if (ans == nums[i])
            count++;
        else
            count--;
    }

    return ans;
}

File: src/_0169_Majority_Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ans = nums[0], count = 0;

        for (int num : nums) {
            if (count == 0)
                ans = num;

            if (ans == num)
                count++;
            else
                count--;
        }

        return ans;
    }
}
